pkg/reconciler/events/cloudevent/cloudevent.go

```diff
@@ -94,22 +96,30 @@ type TektonCloudEventData struct {
 }
 
 // newTektonCloudEventData returns a new instance of TektonCloudEventData
-func newTektonCloudEventData(runObject objectWithCondition) TektonCloudEventData {
+func newTektonCloudEventData(ctx context.Context, runObject objectWithCondition) (TektonCloudEventData, error) {
 	tektonCloudEventData := TektonCloudEventData{}
 	switch v := runObject.(type) {
-	case *v1beta1.TaskRun:
-		tektonCloudEventData.TaskRun = v
-	case *v1beta1.PipelineRun:
-		tektonCloudEventData.PipelineRun = v
```

Comment:
This function works with `objectWithCondition` which is met by both `v1` and `v1beta`.
Ultimately it is invoked through the exported `EmitCloudEvents`, so I think it makes sense to keep the `v1beta1` cases in, since it may be other controllers invoking this.